lib/matplotlib/collections.py

```diff
@@ -1254,6 +1255,244 @@ def set_verts_and_codes(self, verts, codes):
+        """
+        Set new values for the two bounding curves.
+
+        Parameters
+        ----------
+        t : array (length N)
+            The ``self.t_direction`` coordinates of the nodes defining the curves.
+
+        f1 : array (length N) or scalar
+            The other coordinates of the nodes defining the first curve.
+
+        f2 : array (length N) or scalar
+            The other coordinates of the nodes defining the second curve.
+
+        where : array of bool (length N), optional
+            Define *where* to exclude some {dir} regions from being filled.
+            The filled regions are defined by the coordinates ``t[where]``.
+            More precisely, fill between ``t[i]`` and ``t[i+1]`` if
+            ``where[i] and where[i+1]``.  Note that this definition implies
+            that an isolated *True* value between two *False* values in *where*
+            will not result in filling.  Both sides of the *True* position
+            remain unfilled due to the adjacent *False* values.
+
+        See Also
+        --------
+        .PolyCollection.set_verts, .Line2D.set_data
+        """
+        t, f1, f2 = self.axes._fill_between_process_units(
+            self.t_direction, self._f_direction, t, f1, f2)
+
+        verts = self._make_verts(t, f1, f2, where)
+        self.set_verts(verts)
+
+    def get_datalim(self, transData):
+        """Calculate the data limits and return them as a `.Bbox`."""
+        datalim = transforms.Bbox.null()
+        datalim.update_from_data_xy((self.get_transform() - transData).transform(
+            np.concatenate([self._bbox, [self._bbox.minpos]])))
+        return datalim
+
+    def _make_verts(self, t, f1, f2, where):
+        """
+        Make verts that can be forwarded to `.PolyCollection`.
+        """
+        self._validate_dimensions(self.t_direction, self._f_direction, t, f1, f2)
+
+        where = self._normalized_where(t, f1, f2, where)
+        t, f1, f2 = np.broadcast_arrays(np.atleast_1d(t), f1, f2, subok=True)
+        verts = [
+            self._make_verts_for_region(t, f1, f2, idx0, idx1)
+            for idx0, idx1 in cbook.contiguous_regions(where)
+        ]
+
+        self._bbox = transforms.Bbox.null()
+        self._bbox.update_from_data_xy(self._normalize_pts(np.concatenate([
+            np.stack((t[where], f[where]), axis=-1) for f in (f1, f2)])))
+
+        return verts
+
+    def _normalized_where(self, t, f1, f2, where):
+        """Align ``where`` with the masks of ``t``, ``f1`` and ``f2``."""
+        if where is None:
+            where = True
+        else:
+            where = np.asarray(where, dtype=bool)
+            if where.size != t.size:
+                msg = "where size ({}) does not match {!r} size ({})".format(
+                    where.size, self.t_direction, t.size)
+                raise ValueError(msg)
+        return where & ~functools.reduce(
+            np.logical_or, map(np.ma.getmaskarray, [t, f1, f2]))
```

Comment:
Maybe use `np.logical_or.reduce` and drop `functools.reduce`

Reply:
Unfortunately it does not work. The input can be a scalar, where `logical_or` and `functools.reduce` is more tolerant.